File: app/src/main/java/com/jobtick/android/fragments/SignUpFragment.java
package com.jobtick.android.fragments;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import com.google.android.material.button.MaterialButton;
import com.jobtick.android.R;
import android.annotation.SuppressLint;
import android.widget.TextView;

import com.jobtick.android.activities.AuthActivity;
import com.jobtick.android.utils.ExternalIntentHelper;
import com.jobtick.android.utils.Helper;
import com.jobtick.android.widget.ExtendedEntryText;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * A simple {@link Fragment} subclass.
 */
public class SignUpFragment extends Fragment implements AuthActivity.EditTextError {


    @SuppressLint("NonConstantResourceId")
    @BindView(R.id.email)
    ExtendedEntryText edtEmailAddress;
    @SuppressLint("NonConstantResourceId")
    @BindView(R.id.password)
    ExtendedEntryText edtPassword;
    @SuppressLint("NonConstantResourceId")
    @BindView(R.id.confirm_password)
    ExtendedEntryText edtRepeatPassword;

    @SuppressLint("NonConstantResourceId")
    @BindView(R.id.lyt_btn_sign_up)
    MaterialButton lytBtnSignUp;
    @SuppressLint("NonConstantResourceId")
    @BindView(R.id.lyt_btn_google)
    LinearLayout lytBtnGoogle;
    @SuppressLint("NonConstantResourceId")
    @BindView(R.id.lyt_btn_facebook)
    LinearLayout lytBtnFacebook;
    @SuppressLint("NonConstantResourceId")
    @BindView(R.id.txt_btn_sign_in)
    TextView txtBtnSignIn;

    @SuppressLint("NonConstantResourceId")
    @BindView(R.id.txt_btn_terms)
    TextView txtBtnTerms;

    private AuthActivity authActivity;

    private View fragmentView;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        if(fragmentView != null){
            return fragmentView;
        }

        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_sign_up, container, false);
        ButterKnife.bind(this, view);
        authActivity = (AuthActivity) requireActivity();
        if (authActivity != null) {
            authActivity.setEditTextError(this);
        }
        fragmentView = view;
        return view;
    }


    private boolean validation() {
        if (TextUtils.isEmpty(edtEmailAddress.getText().trim())) {
            edtEmailAddress.setError("Check your email address");
            return false;
        }
        else if (TextUtils.isEmpty(edtPassword.getText().trim())) {
            edtPassword.setError("Enter your password");
            return false;
        }else if(edtPassword.getText().trim().length() < 8){
            edtPassword.setError("Password must be atleast 8 characters.");
            return false;
        } else if (!edtPassword.getText().trim().equals(edtRepeatPassword.getText().trim())) {
            edtRepeatPassword.setError("password doesn't match");
            return false;
        }
        return true;
    }


    @OnClick({R.id.lyt_btn_sign_up, R.id.lyt_btn_google, R.id.lyt_btn_facebook, R.id.txt_btn_sign_in, R.id.txt_btn_terms})
    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.lyt_btn_sign_up:
                if (validation()) {
                    authActivity.Signup(edtEmailAddress.getText().trim(), edtPassword.getText().trim());
                }
                break;
            case R.id.lyt_btn_google:
                authActivity.signInWithGoogle(true);
                break;
            case R.id.lyt_btn_facebook:
                authActivity.facebookLogin(true);
                break;
            case R.id.txt_btn_sign_in:
                Helper.closeKeyboard(authActivity);
                Fragment fragment = new SignInFragment();
                FragmentTransaction ft = authActivity.getSupportFragmentManager().beginTransaction();
                ft.replace(R.id.auth_layout, fragment).addToBackStack(SignInFragment.class.getName());
                ft.commit();
                break;
            case R.id.txt_btn_terms:
                ExternalIntentHelper.openLink(requireActivity(), "https://www.jobtick.com/terms");
                break;

        }
    }


    @Override
    public void onEmailError(String emailError) {
        edtEmailAddress.setError(emailError);
    }
    @Override
    public void onPasswordError(String passwordError) {
        edtPassword.setError(passwordError);
    }
}

File: app/src/main/java/com/jobtick/android/fragments/SignUpFragment.kt
package com.jobtick.android.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import androidx.appcompat.widget.AppCompatCheckBox
import androidx.fragment.app.Fragment
import com.google.android.material.button.MaterialButton
import com.jobtick.android.R
import com.jobtick.android.activities.AuthActivity
import com.jobtick.android.activities.AuthActivity.EditTextError
import com.jobtick.android.utils.ExternalIntentHelper
import com.jobtick.android.utils.Helper
import com.jobtick.android.widget.ExtendedEntryText
import com.jobtick.android.widget.ExtendedEntryTextDiffId

/**
 * A simple [Fragment] subclass.
 */
class SignUpFragment : Fragment(), EditTextError {

    private var edtEmailAddress: ExtendedEntryTextDiffId? = null
    private var edtPassword: ExtendedEntryText? = null
    private var edtRepeatPassword: ExtendedEntryText? = null
    private var lytBtnSignUp: MaterialButton? = null
    private var lytBtnGoogle: LinearLayout? = null
    private var lytBtnFacebook: LinearLayout? = null
    private var txtBtnSignIn: TextView? = null
    private var txtBtnTerms: TextView? = null
    private var cbTerms: AppCompatCheckBox? = null
    private var authActivity: AuthActivity? = null
    private var fragmentView: View? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        if (fragmentView != null) {
            return fragmentView
        }
        val view = inflater.inflate(R.layout.fragment_sign_up, container, false)
        fragmentView = view
        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setIDS()
        onViewClick()
        authActivity = requireActivity() as AuthActivity
        if (authActivity != null) {
            authActivity!!.setEditTextError(this)
        }
    }

    private fun setIDS() {
        edtEmailAddress = requireView().findViewById(R.id.email)
        edtPassword = requireView().findViewById(R.id.password)
        edtRepeatPassword = requireView().findViewById(R.id.confirm_password)
        lytBtnSignUp = requireView().findViewById(R.id.lyt_btn_sign_up)
        txtBtnSignIn = requireView().findViewById(R.id.txt_btn_sign_in)
        txtBtnTerms = requireView().findViewById(R.id.txt_btn_terms)
        cbTerms = requireView().findViewById(R.id.cb_terms)
    }

    private fun onViewClick() {
        lytBtnSignUp!!.setOnClickListener {
            authActivity!!.Signup(
                edtEmailAddress!!.text.trim { it <= ' ' },
                edtPassword!!.text.trim { it <= ' ' },
                cbTerms!!.isChecked
            )
        }
        lytBtnFacebook!!.setOnClickListener { authActivity!!.facebookLogin(true) }
        lytBtnGoogle!!.setOnClickListener { authActivity!!.signInWithGoogle(true) }
        txtBtnSignIn!!.setOnClickListener {
            Helper.closeKeyboard(authActivity)
            val fragment: Fragment = SignInFragment()
            val ft = authActivity!!.supportFragmentManager.beginTransaction()
            ft.replace(R.id.auth_layout, fragment)
                .addToBackStack(SignInFragment::class.java.name)
            ft.commit()
        }
        txtBtnTerms!!.setOnClickListener {
            ExternalIntentHelper.openLink(
                requireActivity(),
                "https://www.jobtick.com/terms"
            )
        }
    }

    override fun onEmailError(emailError: String?) {
        edtEmailAddress!!.setError(emailError)
    }

    override fun onPasswordError(passwordError: String?) {
        edtPassword!!.setError(passwordError)
    }
}
